docs: kubectl openebs plugin (#511)
openebs/website: docs/main/user-guides/kubectl-openebs.md

---
id: kubectl-openebs
title: Kubectl OpenEBS Plugin
keywords:
 - Kubectl
 - Plugin
 - Kubectl Plugin
 - Kubectl OpenEBS Plugin
description: A comprehensive guide to the kubectl-openebs plugin, enabling unified management of OpenEBS storages with enhanced features and improved usability.
---
# Kubectl OpenEBS Plugin

## Overview

> The Kubectl OpenEBS Plugin is an OSS plugin that allows you to interact with all OpenEBS storages through a unified interface. This unified plugin is especially beneficial if you have installed a cluster using the OpenEBS umbrella chart. The supported storages include:

Reviewer: ```suggestion
The Kubectl OpenEBS Plugin is a plugin that allows you to interact with OpenEBS storages planes through a unified interface. This unified plugin is especially beneficial if you have installed a cluster using the OpenEBS helm chart. The supported storages include:
```

- Local PV HostPath
- Local PV LVM
- Local PV ZFS
- Replicated PV Mayastor

This plugin integrates the current `kubectl-mayastor` functionality for Replicated PV Mayastor specific commands while introducing newly developed functionality for other storages, inspired by the existing `kubectl-openebs` plugin. The previous `kubectl-openebs` supported only Local PV storages whereas now it extends its capabilities to include Replicated PV Mayastor commands. Also, the plugin supports multiple output formats, such as YAML and JSON.

> By using this plugin, you can gain a simplified and consistent interface to manage multiple storages, with improved output formats, flexible namespace configurations, and robust command options tailored to meet diverse storage needs.

Reviewer 2: you can gain --> you get

## Features

### Unified Plugin for All Storages

The plugin consolidates functionality for all OpenEBS storages under a single interface. You can manage Local PV Hostpath, Local PV LVM, Local PV ZFS, and Replicated PV Mayastor resources without switching between multiple tools or plugins.

> ### Namespace Assumption

Reviewer 2: Can you refer this section from README.md? I have made some changes there.

https://github.com/openebs/openebs/blob/develop/plugin/README.md#namespace-handling

To set the namespace in the current context, use:

```
kubectl config set-context --namespace=<namespace> --current
```

:::note
Changing the context namespace will also affect how other Kubernetes `kubectl` plugins fetch resources if not explicitly set.
:::

For example, if OpenEBS is installed on namespace "storage" then we can set the `ns` on current context using:

```
kubectl config set-context --namespace=storage --current
```

## Command Overview

This section provides a detailed breakdown of the primary `kubectl-openebs` commands. This plugin serves as the central CLI for managing OpenEBS storage resources.

**Command**

```
kubectl openebs
```

**Expected Output**

```
Storage engines supported  

Usage: kubectl-openebs [OPTIONS] <COMMAND>  

Commands:  
  mayastor          Mayastor operations  
  localpv-lvm       LocalPV LVM operations  
  localpv-zfs       LocalPV ZFS operations  
  localpv-hostpath  LocalPV Hostpath operations  
  help              Print this message or the help of the given subcommand(s)  

> Options:  

Reviewer 2: Please include

-k, --kube-config-path <KUBE_CONFIG_PATH>
          Path to kubeconfig file

Reviewer 3: @abhilashshetty04 should we rename it to `--kubeconfig`? To match the kubectl option name? (on the kubectl-openebs)

Reviewer 2: Yes we can.

Reviewer 2: @balaharish7 , Please refer https://github.com/openebs/openebs/blob/develop/plugin/README.md#namespace-handling

I have made changes. `Kube-config-path` is changed to `kubeconfig`

Author: I've changed `Kube-config-path` to `kubeconfig` throughout the documentation.

---
id: kubectl-openebs
title: Kubectl OpenEBS Plugin
keywords:
 - Kubectl
 - Plugin
 - Kubectl Plugin
 - Kubectl OpenEBS Plugin
description: A comprehensive guide to the kubectl-openebs plugin, enabling unified management of OpenEBS storages with enhanced features and improved usability.
---
# Kubectl OpenEBS Plugin

## Overview

The Kubectl OpenEBS Plugin is a plugin that allows you to interact with OpenEBS storages through a unified interface. This unified plugin is especially beneficial if you have installed a cluster using the OpenEBS Helm chart. The supported storages include:

> 

Reviewer 2: Shall we order this based on how commands are ordered?
 Replicated PV Mayastor
 Local PV LVM
 Local PV ZFS
 Local PV HostPath

Author: The commands are ordered in correctly as per this order:
- Local PV Hostpath
- Local PV LVM
- Local PV ZFS
- Replicated PV Mayastor
